### DISCUSSION TITLE
validation_epoch_end or on_train_epoch_end receive extra arguments/data

### DISCUSSION THREAD

**Discussion by congchan** [Lightning-AI/pytorch-lightning#5888] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
#### What is your question?

Basically I need to validate my model after each epoch, but the labels information is a little tricky(mixture of list of tuple of int and string), that it could  not be included in DataLoader. 

however, `validation_epoch_end, validation_step, etc..`  only receives DataLoader output.

Is there any solution? 

I actually dun want to add the raw data into module which make the code difficult to understand.

**Comment by github-actions** [comment]
Hi! thanks for your contribution!, great first issue!
